scripts/setup-user.sh

```diff
@@ -14,24 +14,44 @@ source /scripts/env-data.sh
 
 # Only create credentials if this is a master database
 # Slave database will just mirror from master users
-echo "Setup postgres User:Password"
-echo "postgresql user: $POSTGRES_USER" > /tmp/PGPASSWORD.txt
-echo "postgresql password: $POSTGRES_PASS" >> /tmp/PGPASSWORD.txt
 
-# Check user already exists
-echo "Creating superuser $POSTGRES_USER"
+echo "$POSTGRES_PASS" >> /tmp/PGPASSWORD.txt
+# Check super user already exists
 RESULT=`su - postgres -c "psql postgres -t -c \"SELECT 1 FROM pg_roles WHERE rolname = '$POSTGRES_USER'\""`
 COMMAND="ALTER"
 if [ -z "$RESULT" ]; then
 	COMMAND="CREATE"
 fi
-su - postgres -c "psql postgres -c \"$COMMAND USER $POSTGRES_USER WITH SUPERUSER ENCRYPTED PASSWORD '$POSTGRES_PASS';\""
 
-echo "Creating replication user $REPLICATION_USER"
+echo "Creating superuser user $POSTGRES_USER using $PASSWORD_AUTHENTICATION authentication "
+if [ PASSWORD_AUTHENTICATION="md5" ]; then
+	PG_PASS=$(U=$POSTGRES_USER; P=$(cat /tmp/PGPASSWORD.txt); echo -n md5; echo -n $P$U | md5sum | cut -d' ' -f1)
+	su - postgres -c "psql postgres -c \"$COMMAND USER $POSTGRES_USER WITH SUPERUSER  PASSWORD '$PG_PASS';\""
```

Comment:
I don't understand this two line.
You append password and user together using `echo -n $P$U` then pass it to md5sum and extract the digest.
This digest is then used as password.
Wouldn't that make the digest the password instead?

Reply:
@lucernae  
When you use use 
```
CREATE USER docker WITH SUPERUSER ENCRYPTED PASSWORD 'foo';
```
 The encryption algorithm that is used for the password will depend on the cluster initialization parameters. In this case we only using md5 or sha256.
If you pass an encrypted password Postgres is smart to figure out that the password is already encrypted and will store it as.
To test try the following 
```
CREATE USER kartoza with password `md5a847d0f7cda274f87440e9e9a3312394`;
``` 
Then log in using the credentials user=`kartoza`;pass=`kartoza`

Reply:
ok, noted I will test it.